pkg/services/ngalert/state/state.go

```diff
@@ -187,6 +187,10 @@ func NewEvaluationValues(m map[string]eval.NumberValueCapture) map[string]*float
 func resultNormal(state *State, _ *models.AlertRule, result eval.Result, logger log.Logger) {
 	if state.State == eval.Normal {
 		logger.Debug("Keeping state", "state", state.State)
+		if state.StartsAt.IsZero() {
```

Comment:
I wonder if it's better to set this in cache.getOrCreate https://github.com/grafana/grafana/blob/0c8876c3a2dcf3bf88a83f11be3e676a2e3b133f/pkg/services/ngalert/state/cache.go#L140-L142